compiler/rustc_mir_build/src/build/coverageinfo.rs

```diff
@@ -1,31 +1,54 @@
+mod mcdc;
 use std::assert_matches::assert_matches;
 use std::collections::hash_map::Entry;
-use std::collections::VecDeque;
 
 use rustc_data_structures::fx::FxHashMap;
-use rustc_middle::mir::coverage::{
-    BlockMarkerId, BranchSpan, ConditionId, ConditionInfo, CoverageKind, MCDCBranchSpan,
-    MCDCDecisionSpan,
-};
+use rustc_middle::mir::coverage::{BlockMarkerId, BranchSpan, CoverageKind};
 use rustc_middle::mir::{self, BasicBlock, SourceInfo, UnOp};
-use rustc_middle::thir::{ExprId, ExprKind, LogicalOp, Thir};
+use rustc_middle::thir::{ExprId, ExprKind, Thir};
 use rustc_middle::ty::TyCtxt;
 use rustc_span::def_id::LocalDefId;
-use rustc_span::Span;
 
 use crate::build::{Builder, CFG};
-use crate::errors::MCDCExceedsConditionNumLimit;
+use mcdc::MCDCInfoBuilder;
```

Comment:
Nit: Please make this `use crate::build::coverageinfo::mcdc::MCDCInfoBuilder`, so that all the same-crate imports are consistent.

(I don't think there's an official policy for this formatting, but personally I prefer the consistent style.)